File: src/main/java/com/lessmarkup/interfaces/structure/NodeCache.java
package com.lessmarkup.interfaces.structure;

import com.lessmarkup.interfaces.cache.CacheHandler;
import java.util.List;

public interface NodeCache extends CacheHandler {
    CachedNodeInformation getNode(long nodeId);
    Tuple<CachedNodeInformation, String> getNode(String path);
    CachedNodeInformation getRootNode();
    List<CachedNodeInformation> getNodes();
    NodeHandler getNodeHandler(String path, GetNodeHandlerPreprocess filter);
    default NodeHandler getNodeHandler(String path) {
        return getNodeHandler(path, null);
    }
}

File: src/main/scala/com/lessmarkup/interfaces/structure/NodeCache.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.structure

import com.lessmarkup.interfaces.cache.CacheHandler

trait NodeCache extends CacheHandler {
  def getNode(nodeId: Long): Option[CachedNodeInformation]

  def getNode(path: String): Option[(CachedNodeInformation, String)]

  def getRootNode: Option[CachedNodeInformation]

  def getNodes: Seq[CachedNodeInformation]

  def getNodeHandler(path: String, filter: Option[(NodeHandler, String, String, Seq[String], Option[Long]) => Boolean] = None): Option[NodeHandler]

  def getNodeHandler(path: String): Option[NodeHandler] = {
    getNodeHandler(path, null)
  }
}
